### DISCUSSION TITLE
Support for YAML imports (or support for being able to do it)

### DISCUSSION THREAD

**Discussion by a-y-u-s-h** [remix-run/remix#7549] [General]
I want to be able to import YAML files inside my components as JS objects because I have statecharts described mostly in YAML and implemented in JS or TS. I can very well convert them into JSON and use them, but YAML is better for big structures because it is easier to change and I can comment in it as much as I want (JSON doesn't support comments and JS objects will be relatively poor readability + speed of development compared to what YAML gives me with statecharts). Either way, a person can have several reasons to import YAML files.

In NextJS I would simply add a `yaml-loader` in the config and it would just work. YAML is just like JSON and I'm not asking for it to be supported natively, but maybe a way to add custom-loaders would be the way to go. Currently, this is the only thing that's stopping me from porting all of my old applications to Remix. There are a few more things that I am yet to figure out while combining statecharts with Remix, but I'll ask about them later. 

I dug around and found out that this feature is not supported yet (from somebody's query in discord around 8 months ago) so I'm opening up an issue/discussion about it.

**Comment by arnaudbzn** [comment]
Since Remix 2.2.0 and Vite support you can use  `@rollup/plugin-yaml` vite plugin.

Setup from [Astro doc](https://docs.astro.build/en/recipes/add-yaml-support/)
```
npm install @rollup/plugin-yaml --save-dev
```

vite.config.ts: Add yaml plugin
```
import { unstable_vitePlugin as remix } from "@remix-run/dev";
import { defineConfig } from "vite";
import yaml from '@rollup/plugin-yaml';

export default defineConfig({
  vite: {
    plugins: [remix(),yaml()]
  }
});
```

env.d.ts add:
```
declare module "*.yml" {
  const value: any; // Add type definitions here if desired
  export default value;
}
```

Anywhere in your app:
```
import yaml from "./data.yml";
```

**Reply by sergiodxa** [reply]
I would suggest typing value as unknown instead
```ts
declare module "*.yml" {
  const value: unknown; // Add type definitions here if desired
  export default value;
}
```
So you're forced to narrow the type on runtime.
